ext/fg/js/frontend.js

```diff
@@ -71,162 +62,9 @@ class Frontend {
-                    // Aborted
-                    return;
-                }
-            }
-
-            await this.searchAt(e.clientX, e.clientY, 'mouse');
-        };
-
-        search();
-    }
-
-    onMouseDown(e) {
-        if (this.preventNextMouseDown) {
-            this.preventNextMouseDown = false;
-            this.preventNextClick = true;
-            e.preventDefault();
-            e.stopPropagation();
-            return false;
-        }
-
-        if (e.button === 0) {
-            this.popupTimerClear();
-            this.searchClear(true);
-        }
-    }
-
-    onMouseOut() {
-        this.popupTimerClear();
-    }
-
-    onClick(e) {
-        if (this.preventNextClick) {
-            this.preventNextClick = false;
-            e.preventDefault();
-            e.stopPropagation();
-            return false;
-        }
-    }
-
-    onAuxClick() {
-        this.preventNextContextMenu = false;
-    }
-
-    onContextMenu(e) {
-        if (this.preventNextContextMenu) {
-            this.preventNextContextMenu = false;
-            e.preventDefault();
-            e.stopPropagation();
-            return false;
-        }
-    }
-
-    onTouchStart(e) {
-        if (this.primaryTouchIdentifier !== null || e.changedTouches.length === 0) {
-            return;
-        }
-
-        this.preventScroll = false;
-        this.preventNextContextMenu = false;
-        this.preventNextMouseDown = false;
-        this.preventNextClick = false;
-
-        const primaryTouch = e.changedTouches[0];
-        if (DOM.isPointInSelection(primaryTouch.clientX, primaryTouch.clientY, window.getSelection())) {
-            return;
-        }
-
-        this.primaryTouchIdentifier = primaryTouch.identifier;
-
-        if (this.pendingLookup) {
-            return;
-        }
-
-        const textSourceCurrentPrevious = this.textSourceCurrent !== null ? this.textSourceCurrent.clone() : null;
-
-        this.searchAt(primaryTouch.clientX, primaryTouch.clientY, 'touchStart')
-        .then(() => {
-            if (
-                this.textSourceCurrent === null ||
-                this.textSourceCurrent.equals(textSourceCurrentPrevious)
-            ) {
-                return;
-            }
-
-            this.preventScroll = true;
-            this.preventNextContextMenu = true;
-            this.preventNextMouseDown = true;
-        });
-    }
-
-    onTouchEnd(e) {
-        if (
-            this.primaryTouchIdentifier === null ||
-            this.getIndexOfTouch(e.changedTouches, this.primaryTouchIdentifier) < 0
-        ) {
-            return;
-        }
-
-        this.primaryTouchIdentifier = null;
-        this.preventScroll = false;
-        this.preventNextClick = false;
-        // Don't revert context menu and mouse down prevention,
-        // since these events can occur after the touch has ended.
-        // this.preventNextContextMenu = false;
-        // this.preventNextMouseDown = false;
```

Comment:
Was this commented-out code here on purpose? The explanation above looks like it was, but I'm not sure. I just copied it over to the new class.

Reply:
I believe the code was commented on purpose, with the comment indicating that it's necessary to _not_ run that code.